closure/goog/structs/avltree.js

```diff
@@ -898,4 +899,47 @@ Node.prototype.fixHeight = function() {
       1;
 };
 
+/**
+ * Returns a deep copy of the tree in O(n * k),
+ * where k is the time complexity to copy a node.
+ * @return {AvlTree}
+ */
+AvlTree.prototype.copy = function() {
+  const tree = new AvlTree();
+
+  // Copy instance properties
+  const {copy, leftMost, rightMost} = this.root_.copy();
+  tree.root_       = copy;
+  tree.comparator_ = this.comparator_;
+  tree.minNode_    = leftMost;
+  tree.maxNode_    = rightMost;
+
+  return tree;
+};
+
+Node.prototype.copy = function(parent) {
+  const val  = JSON.parse(JSON.stringify(this.value)); // deep copy
+  const node = new Node(val, parent);
+
+  // Copy all properties
+  node.count  = this.count;
+  node.height = this.height;
+
+  var minNode = node, maxNode = node;
```

Comment:
If you want to use ES6 in this new code (which I'm not sure will land - but I'll fix it when I pull this in if it can't) then please be consistent and use `const` or `let` here..